### ISSUE TITLE
The relabel module does not allow "\n" inside a character class

### ISSUE THREAD

**Issue by hugomg** [sqmedeiros/lpeglabel#25]
LPeg allows "\n" inside a character class:

```
$ lua -lre
Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
> re.compile("[\n]")
lpeg-pattern: 0x5576faf9b888
```

However, LPegLabel does not:

```
$ lua -lrelabel
Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
> relabel.compile("[\n]")
syntax error(s) in pattern
L1:C2: expected at least one item after '[' or '^'
[
 ^
stack traceback:
	[C]: in function 'error'
	/home/hugo/.luarocks/share/lua/5.3/relabel.lua:317: in function 'relabel.compile'
	(...tail calls...)
	[C]: in ?
```

We can work around the problem by using `%nl` instead of `\n` but I think it would be nice if we could use `\n` inside a character class. I ran these tests on LPeg 1.0.2-1 and LPegLabel 1.6.0-1

**Comment by sqmedeiros** [comment]
Hi, Hugo.
Matthew, GSoC 2016 student, originally adapted 're' to use labels as part of his GSoC tasks.
I think '\n' is not allowed inside a character class (and also inside a string) because Matthew tried to implement error recovery, and used ';' and '\n' as synchronization points during recovery.
As currently 'relabel' does not recover from a syntactical error, I think we can accept '\n' inside a character class or a string.
I created a branch [newline](https://github.com/sqmedeiros/lpeglabel/tree/newline) to address this issue. Please, try it and let me know if it works as expected.

**Comment by hugomg** [comment]
Thanks! It now works as expected.

**Comment by sqmedeiros** [comment]
Fixed by PR #29
